### PR TITLE
Add rules for nft rewards

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2156** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2156" target="_blank">Graphite</a> 
    * **PR #2169** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2169" target="_blank">Graphite</a> 
      * **PR #2170** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2170" target="_blank">Graphite</a> 
        * **PR #2171** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2171" target="_blank">Graphite</a> 
          * **PR #2172** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2172" target="_blank">Graphite</a> 
            * **PR #2181** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2181" target="_blank">Graphite</a> 
              * **PR #2182** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2182" target="_blank">Graphite</a> 
                * **PR #2183** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2183" target="_blank">Graphite</a> 
                  * **PR #2184** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2184" target="_blank">Graphite</a> 
                    * **PR #2185** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2185" target="_blank">Graphite</a> 
                      * **PR #2186** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2186" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2186?utm_source=stack-comment).

File: src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export const inputIsValidUrlRule = (props: {
+  label?: string
+  prefix?: string
+  validateTrigger?: string
+}) => ({
+  validator: (rule: RuleObject, value: unknown) => {
+    if (value === undefined) return Promise.resolve()
+    if (typeof value !== 'string')
+      return Promise.reject(t`Invalid type - contact Juicebox Support`)
```
Comment (tomquirk) [inline src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts:15]: **suggestion** Idk if contact Juicebox Support is an appropriate CTA. Is there any reason value wouldn't be a string, other than a bug?

File: src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    let string = value
+    if (props.prefix) {
+      string = props.prefix + value
+    }
+    try {
+      new URL(string)
+      return Promise.resolve()
+    } catch {
+      return Promise.reject(
+        props.label ? t`${props.label} is not valid URL` : `Is not valid URL`,
```
Comment (tomquirk) [inline src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts:25]:
```suggestion
        props.label ? t`${props.label} is not a valid URL` : `Is not a valid URL`,
```

File: src/components/Create/components/pages/utils/rules/inputNonZeroRule.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { t } from '@lingui/macro'
+import { RuleObject } from 'antd/lib/form'
+
+const nonNumericRejection = (label: string | undefined) => {
```
Comment (tomquirk) [inline src/components/Create/components/pages/utils/rules/inputNonZeroRule.ts:4]: **praise ** love all the rules colocated in the same directory <3

**Review by tomquirk — APPROVED** [review]

File: src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts (hunk 1 above)
Comment (wraeth-eth) [reply src/components/Create/components/pages/utils/rules/inputIsValidUrlRule.ts:15]: doesn't make sense for it to be anything other than a string - would just be a bug

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2186?utm_source=stack-comment).


Job Reference: BcMq9niLYsok9Kx1gVMU

**Pull request merged**